Db2csv

mkutu opened the pull request:
refactors mask_gen to save to csv instead of individual jsons and updatews the LTS database using the temporary local database (csv). 

Todo include:
- [x] refactoring detect.py to output into a single csv
- [x] use csv from detect.py in segment.py
- [x] use the csv in inspect.py creating the necessary columns and updating tags
- [x] implement csv in refine.py to record refine_params, and new mask locations, update reviewer, status
- [x] implement csv in export_cvat.py by updating tags and adding new columns with relabeled_mask location, update status, and reviewer
- [x] make changes to fields based on https://github.com/precision-sustainable-ag/Field-AnnotationPipeline/issues/112
- [x] update the 'cropout' using the newly relabeled mask in `export_cvat.py`
---
- [x] **Move data to LTS** including:
  - [x] move fullsized masks to `/mnt/research-projects/r/raatwell/longterm_images3/field-batches/masks/`
  - [x] move cutouts to `/mnt/research-projects/r/raatwell/longterm_images3/field-batches/cutouts/`
- [x] **Update DB by matching `image_ids`**
    - [x] Update `mask_timestamp`
    - [x] Update `refine_params`
    - [x] Update `seg_note`
    - [x] Update `detection_note`
    - [x] Set `mask_status` → "finalized"
    - [x] Update `temp_relabeled_full_mask_path` → `final_mask_path`
    - [x] Update `temp_relabeled_cutout_image_path` → `cutout_image_path`
    - [x] Remove `mask_review_datetime`
    - [x] Update `mask_reviewer`
    - [x] Update `final_mask_issue_tag`
    - [x] Remove `refined_params` (duplicate of `refine_params`)
---
- [ ] test the entire pipeline:
  - [x] `mask_gen.create_project`
  - [x] `mask_gen.detect`
  - [x] `mask_gen.segment`
  - [x] `mask_gen.inspect`
  - [x] `mask_gen.refine`
  - [x] `mask_gen.upload_cvat`
  - [ ] `mask_gen.export_cvat`
  - [ ] `finalize.finalize_lts`

navjot-nangia committed:
refactored missing_white to use luminance index

sbaghba committed:
added base augmentation implementation

mkutu committed:
initial working skeleton for relabeling

mkutu committed:
renamed files

navjot-nangia committed:
added luminance parameter to work on missing_white

navjot-nangia committed:
corrected method name in missing_white

sbaghba committed:
added visualization flag to main config - moved use flag to train config

sbaghba committed:
minor corrections

sbaghba committed:
improved augmentation visualization

sbaghba committed:
added resize to gridcrop

sbaghba committed:
preprocess directory changed

sbaghba committed:
implemented most of straightforward transformations

mkutu committed:
refactor relabeling

mkutu committed:
fixes for inital_tag and refine_params handling

mkutu committed:
Add export_cvat functionality and enhance relabeling configuration

mkutu committed:
Handle missing initial_tag in sample data to prevent KeyError

mkutu committed:
name change from relabel to upload_cvat.py

mkutu committed:
set relabeled_path

sbaghba committed:
added "some of" concept

sbaghba committed:
general fixes esp dropout + batch augs

sbaghba committed:
better viz

mkutu committed:
Update .gitignore and configuration files for improved data handling and export settings

sbaghba committed:
added output type annotations

mkutu committed:
Merge pull request #92 from precision-sustainable-ag/missing_white_refine

added luminance parameter to work on missing_white

mkutu committed:
add species labeling to db

sbaghba committed:
fixed path issues

mkutu committed:
skeleton structure for creating project by filtering by species

mkutu committed:
skeleton structure for creating project by filtering by species

sbaghba committed:
added "some of"

sbaghba committed:
opted for simpler print. the alert emails the user.

sbaghba committed:
better docstrings / fixed visualization and batch aug logics

navjot-nangia committed:
Refactor project creation and filtering by species; add new configuration files and update paths

navjot-nangia committed:
Refactor project creation and filtering by species; add new configuration files and update paths

navjot-nangia committed:
Merge branch 'develop' into feature/create_project

mkutu committed:
cleaned up entry point

mkutu committed:
cleaned up entry point

mkutu committed:
simplified project creation and image copying using new db column

mkutu committed:
simplified project creation and image copying using new db column

mkutu committed:
simplified sampling and clean connecting and closing of db

mkutu committed:
simplified sampling and clean connecting and closing of db

mkutu committed:
removed unused  paths in config

mkutu committed:
removed unused  paths in config

mkutu committed:
Add CSV output for detection results

mkutu committed:
Add CSV output for detection results

mkutu committed:
Add CSV output for detection results

mkutu committed:
Integrate CSV handling for detections in UNetInference, updating mask paths and processing images from DataFrame and saving back

mkutu committed:
Integrate CSV handling for detections in UNetInference, updating mask paths and processing images from DataFrame and saving back

mkutu committed:
Integrate CSV handling for detections in UNetInference, updating mask paths and processing images from DataFrame and saving back

mkutu committed:
fix: correct logging message and enhance image filtering in database query

mkutu committed:
fix: correct logging message and enhance image filtering in database query

mkutu committed:
filter out without mat and color checker images from sample

mkutu committed:
filter out without mat and color checker images from sample

mkutu committed:
Skeleton filter_by_species.py changes. Non-workling changes. filter_by_species.py needs updating.

mkutu committed:
Skeleton filter_by_species.py changes. Non-workling changes. filter_by_species.py needs updating.

sbaghba committed:
file and folder restructure

sbaghba committed:
image_logs in naming

sbaghba committed:
data folder

navjot-nangia committed:
- updated the DataFrame with the local path using the image_id index
- updated documentation
- updated logging

navjot-nangia committed:
- updated the DataFrame with the local path using the image_id index
- updated documentation
- updated logging

navjot-nangia committed:
removed donwnscaled images as we are working with full scale images now

navjot-nangia committed:
removed donwnscaled images as we are working with full scale images now

navjot-nangia committed:
- minor doc changes
- removed cash and covercrops from the yaml file to track species
- updated yaml file for species tracking

navjot-nangia committed:
- minor doc changes
- removed cash and covercrops from the yaml file to track species
- updated yaml file for species tracking

navjot-nangia committed:
- minor

navjot-nangia committed:
- minor

navjot-nangia committed:
minor

navjot-nangia committed:
minor

mkutu committed:
changed column path to local_developed_image_path

mkutu committed:
changed column path to local_developed_image_path

mkutu committed:
minor fix: simplify DataFrame handling and improve image filtering logic

mkutu committed:
minor fix: simplify DataFrame handling and improve image filtering logic

mkutu committed:
Merge pull request #98 from precision-sustainable-ag/feature/create_project

Feature/create project

mkutu committed:
Merge branch 'db2csv' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into db2csv

mkutu committed:
Refactor weed detection processing to load from CSV and resolve image paths dynamically

mkutu committed:
Refactor weed detection processing to load from CSV and resolve image paths dynamically

mkutu committed:
Refactor segment.py to use temp_db

mkutu committed:
Refactor segment.py to use temp_db

mkutu committed:
Add project_temp_db path and moved it's save location

mkutu committed:
Add project_temp_db path and moved it's save location

mkutu committed:
made create_project a subtask of mask_gen

mkutu committed:
made create_project a subtask of mask_gen

mkutu committed:
implement temp db in inspect.py

mkutu committed:
implement temp db in inspect.py

mkutu committed:
minor fix: removed unecessary columns

mkutu committed:
minor fix: removed unecessary columns

mkutu committed:
implement temp db to refine.py

mkutu committed:
implement temp db to refine.py

sbaghba committed:
pt 1 and 2 are addressed

sbaghba committed:
addressed all points except albumentation restructure - kept plots in the previous dir as they were

mkutu committed:
Refactor upload_cvat.py to streamline CSV handling and enhance CVAT upload process

mkutu committed:
Refactor upload_cvat.py to streamline CSV handling and enhance CVAT upload process

navjot-nangia committed:
Update species_images in YAML configuration files of create project.

navjot-nangia committed:
Update species_images in YAML configuration files of create project.

navjot-nangia committed:
minor

navjot-nangia committed:
minor

navjot-nangia committed:
minor

navjot-nangia committed:
minor

navjot-nangia committed:
minor

navjot-nangia committed:
minor

navjot-nangia committed:
making sure the values are of type "object". A "FutureWarning" message was showing up while running the script.

navjot-nangia committed:
making sure the values are of type "object". A "FutureWarning" message was showing up while running the script.

navjot-nangia committed:
added some logging

navjot-nangia committed:
added some logging

navjot-nangia commented:
@mkutu 
DONE: Easy fix
~~Just realized that, in inspection task, the "finalized" mask status is not getting reviewer name in the temp db. Will try to resolve it.~~ 
<img width="1567" height="529" alt="Screenshot 2025-08-13 153730" src="https://github.com/user-attachments/assets/3f90bbc8-1dc9-433c-822d-a0f00951000b" />

navjot-nangia committed:
making sure that the "finalized" mask_status also get the "mask_reviewer" info

navjot-nangia committed:
making sure that the "finalized" mask_status also get the "mask_reviewer" info

navjot-nangia commented:
@mkutu 
We can discuss in meeting tomorrow how should this be done:

- [ ] in temp_db, the refined mask path is missing for finalized mask_status
- [ ] final_mask_issue_tag (eg: good) and mask_status (eg: finalized) feels redundant

<img width="1233" height="533" alt="Screenshot 2025-08-13 155915" src="https://github.com/user-attachments/assets/930a8bfe-4179-46f4-bb4f-23299528406d" />
<img width="969" height="525" alt="Screenshot 2025-08-13 155901" src="https://github.com/user-attachments/assets/a4d3841e-b3ab-449b-a39f-135fd8ab04a0" />

sbaghba committed:
albumentation suggestion implemented - missing domain specifics and normalization

sbaghba committed:
added new augmentations and domain specific - missing normalization

sbaghba committed:
fixed visualizer with correct augs displayed

sbaghba committed:
applied random_grid_shuffle correction

sbaghba committed:
part A Done

mkutu committed:
update field name changes in inspect.py

mkutu committed:
update field name changes in inspect.py

mkutu committed:
update column name changes

mkutu committed:
update column name changes

mkutu committed:
Enhance image filtering in FilterImagesBySpecies by using a temporary column for extension checks

mkutu committed:
Enhance image filtering in FilterImagesBySpecies by using a temporary column for extension checks

mkutu committed:
Refactor MissingWhite to improve speckle pruning

mkutu committed:
Refactor MissingWhite to improve speckle pruning

mkutu committed:
Refactor export_cvat.py to enhance CVAT relabeling process and improve mask handling

mkutu committed:
Refactor export_cvat.py to enhance CVAT relabeling process and improve mask handling

mkutu committed:
Merge branch 'db2csv' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into db2csv

mkutu committed:
Add functionality to create masked cutout images from relabeled masks

mkutu committed:
Add functionality to create masked cutout images from relabeled masks

mkutu committed:
mode=finalize step that moves masks and cutouts to and updated db in lts.

mkutu committed:
mode=finalize step that moves masks and cutouts to and updated db in lts.

mkutu committed:
db field update changes

mkutu committed:
db field update changes

mkutu committed:
old cfg update from missing white update

mkutu committed:
old cfg update from missing white update

mkutu committed:
minor path fixes

mkutu committed:
minor path fixes

mkutu committed:
Add finalize settings to configuration and improve image filtering logic

mkutu committed:
Add finalize settings to configuration and improve image filtering logic

sbaghba committed:
tested spatial - pixels remaining

sbaghba committed:
pixels are done

sbaghba committed:
added image wide and channel wide normalization

mkutu committed:
add flder and keys to gitignore

navjot-nangia committed:
species distribution for each user

navjot-nangia committed:
species distribution for each user

mkutu committed:
Merge branch 'db2csv' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into db2csv

mkutu committed:
Enhance mask finalization logic and add file existence checks

mkutu committed:
Enhance mask finalization logic and add file existence checks

mkutu committed:
Merge branch 'HAugment' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into HAugment

mkutu committed:
augment argument adjustments (minor)

mkutu committed:
fixed logic and added both augment and batch visualizer

mkutu committed:
update .gitignore to include notebooks directory and set train_pipeline to false in config.yaml

mkutu committed:
fixed train_pipeline viz logic and minor doc for nonimplemented augments

mkutu committed:
- Changed the database file name in default.yaml for clarity.
- Added handling for NaN values in initial and reviewer tags in export_cvat.py.

mkutu committed:
- Changed the database file name in default.yaml for clarity.
- Added handling for NaN values in initial and reviewer tags in export_cvat.py.

sbaghba committed:
fixed wandb logging

mkutu committed:
Add check to skip rows without detection results in CVAT upload

mkutu committed:
Add check to skip rows without detection results in CVAT upload

mkutu committed:
improved logic

mkutu committed:
improved logic

navjot-nangia committed:
Fixed mask status empty issue

navjot-nangia committed:
Fixed mask status empty issue

mkutu committed:
updated cvat relabeling task name

mkutu committed:
updated cvat relabeling task name

navjot-nangia committed:
Merge pull request #104 from precision-sustainable-ag/fill-no-detection

Add check to skip rows without detection results in CVAT upload

navjot-nangia committed:
- added 'unreviewed' to ignore in mask_status
- made sure the "initial_tag" and "reviewer" from fiftyone samples is handled correctly

navjot-nangia committed:
minor

navjot-nangia committed:
minor: changed relabelled to relabeled for consistency

mkutu committed:
Merge branch 'develop' into HAugment

mkutu committed:
Merge pull request #89 from precision-sustainable-ag/HAugment

Augmentation PR

mkutu committed:
Merge branch 'feature/relabel' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into feature/relabel

mkutu committed:
Merge branch 'db2csv' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into db2csv

mkutu committed:
Merge branch 'finalize-lts-masks' of https://github.com/precision-sustainable-ag/Field-SegmentationTraining into finalize-lts-masks

navjot-nangia committed:
minor for consistency

navjot-nangia committed:
Ensure string conversion for mask issue tags and metadata in CVAT sample creation

navjot-nangia committed:
minor: for consistency

navjot-nangia committed:
Ensure string conversion for mask issue tags and metadata in CVAT sample creation

navjot-nangia committed:
marking 'finalized' for 'final_mask_tag' if inspection tag is 'good'

navjot-nangia committed:
- Fixed typo in STATUS_RELABELLED to STATUS_RELABELED
- made sure nan values in final_tag are taken care of

navjot-nangia committed:
minor

navjot-nangia committed:
Fix typo in mask status from 'relabelled' to 'relabeled'

navjot-nangia committed:
Fix typo in mask status from 'relabelled' to 'relabeled'

mkutu committed:
Refactor mask generation paths and update related methods for clarity and consistency

mkutu committed:
Merge branch 'db2csv' into finalize-lts-masks

mkutu committed:
Merge pull request #102 from precision-sustainable-ag/finalize-lts-masks

Finalize lts masks
